packages/combobox/src/LionCombobox.js

```diff
@@ -413,54 +428,62 @@ export class LionCombobox extends OverlayMixin(LionListbox) {
    *   return options.currentValue.length > 4 && super._showOverlayCondition(options);
    * }
    *
-   * @param {{ currentValue: string, lastKey:string|undefined }} options
+   * @param {{ currentValue?: string, lastKey?: string }} options
    * @protected
    * @returns {boolean}
    */
+  // TODO: batch all pending condition triggers in __pendingShowTriggers, reducing race conditions
   // eslint-disable-next-line class-methods-use-this
   _showOverlayCondition({ lastKey }) {
+    const hideOn = ['Tab', 'Escape', 'Enter'];
+    if (lastKey && hideOn.includes(lastKey)) {
+      return false;
+    }
+
     if (this.showAllOnEmpty && this.focused) {
       return true;
     }
     // when no keyboard action involved (on focused change), return current opened state
     if (!lastKey) {
       return /** @type {boolean} */ (this.opened);
     }
-    const doNotShowOn = ['Tab', 'Esc', 'Enter'];
-    return !doNotShowOn.includes(lastKey);
+    return true;
   }
 
   /**
-   * @param {Event} ev
+   * @configure ListboxMixin whenever the options are changed (potentially due to external causes
+   * like server side filtering of nodes), schedule autocompletion for proper highlighting
    * @protected
    */
+  _onListboxContentChanged() {
+    super._onListboxContentChanged();
+    this.__shouldAutocompleteNextUpdate = true;
+  }
+
   // eslint-disable-next-line no-unused-vars
-  _textboxOnInput(ev) {
-    // Schedules autocompletion of options
+  _textboxOnInput() {
     this.__shouldAutocompleteNextUpdate = true;
   }
 
   /**
    * @param {KeyboardEvent} ev
    * @protected
    */
-  _textboxOnKeydown(ev) {
-    if (ev.key === 'Tab') {
-      this.opened = false;
-    }
-  }
+  // eslint-disable-next-line class-methods-use-this, no-unused-vars
+  _textboxOnKeydown(ev) {}
```

Comment:
Before the overlay closed when tabbing out of it, why did you remove this feature?

Reply:
It was already checked here: https://github.com/ing-bank/lion/pull/1341/files/22678f13ce450e43fcdc94c49456ad37073abd43#diff-03152156fa1a67197eb600827fa4d55e5cf2571242b96a7333935109c12cbe33R438

Reply:
Correction: you were right indeed 🙂 Great catch. (apparently we need to check this on keydown and not on keyup and the test wrongly sent a keyup event, which in non apparent in the browser)

Reply:
See: https://github.com/ing-bank/lion/pull/1341/commits/788f500a2ca38855ebcbf70db30b015fb667fa69